### ISSUE TITLE
shift+click+drag doesnt delete blocks in nightly editor

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#5705]
reported by louis

**Comment by def-** [comment]
On 16.2.2 it "works", but crashes sometimes:
```
==37463==ERROR: AddressSanitizer: container-overflow on address 0x00012b42c9d0 at pc 0x000101bf5cd8 bp 0x00016f68a150 sp 0x00016f68a148
READ of size 8 at 0x00012b42c9d0 thread T0
    #0 0x101bf5cd4 in CEditor::DoMapEditor(CUIRect) editor.cpp:2639
    #1 0x101c8eee0 in CEditor::Render() editor.cpp:5773
    #2 0x101caf9d4 in CEditor::OnRender() editor.cpp:6463
    #3 0x100ca092c in CClient::Run() client.cpp:3330
    #4 0x100cf3f2c in TWMain client.cpp:4646
    #5 0x101094e1c in main client.mm:21
    #6 0x103521088 in start+0x204 (dyld:arm64e+0x5088)

0x00012b42c9d0 is located 0 bytes inside of 8-byte region [0x00012b42c9d0,0x00012b42c9d8)
allocated by thread T0 here:
    #0 0x10573cbd8 in wrap__Znwm+0x74 (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x4cbd8)
    #1 0x101cc0874 in void* std::__1::__libcpp_operator_new<unsigned long>(unsigned long) new:235
    #2 0x101cc0818 in std::__1::__libcpp_allocate(unsigned long, unsigned long) new:261
    #3 0x101cf8518 in std::__1::allocator<CLayer*>::allocate(unsigned long) allocator.h:108
    #4 0x101cf7f5c in std::__1::allocator_traits<std::__1::allocator<CLayer*> >::allocate(std::__1::allocator<CLayer*>&, unsigned long) allocator_traits.h:262
    #5 0x101d0e148 in std::__1::__split_buffer<CLayer*, std::__1::allocator<CLayer*>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<CLayer*>&) __split_buffer:315
    #6 0x101d0d910 in std::__1::__split_buffer<CLayer*, std::__1::allocator<CLayer*>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<CLayer*>&) __split_buffer:314
    #7 0x101d0d2cc in void std::__1::vector<CLayer*, std::__1::allocator<CLayer*> >::__push_back_slow_path<CLayer* const&>(CLayer* const&) vector:1646
    #8 0x101b6fbf4 in std::__1::vector<CLayer*, std::__1::allocator<CLayer*> >::push_back(CLayer* const&) vector:1663
    #9 0x101b6f960 in CLayerGroup::AddLayer(CLayer*) editor.cpp:204
    #10 0x101e4a6f8 in CLayerTiles::BrushGrab(CLayerGroup*, CUIRect) layer_tiles.cpp:398
    #11 0x101bf4a64 in CEditor::DoMapEditor(CUIRect) editor.cpp:2611
    #12 0x101c8eee0 in CEditor::Render() editor.cpp:5773
    #13 0x101caf9d4 in CEditor::OnRender() editor.cpp:6463
    #14 0x100ca092c in CClient::Run() client.cpp:3330
    #15 0x100cf3f2c in TWMain client.cpp:4646
    #16 0x101094e1c in main client.mm:21
    #17 0x103521088 in start+0x204 (dyld:arm64e+0x5088)

HINT: if you don't care about these errors you may set ASAN_OPTIONS=detect_container_overflow=0.
```

**Comment by def-** [comment]
Broken/fixed by de087bd80a
